### PR TITLE
tests: enable testing with snapd snap FIPS variant

### PR DESCRIPTION
**Pull request by bboozzoo**
Based on #14439. Add spread testing of FIPS snap variant.

Related: SNAPDENG-23245

### CODE REVIEW

File: build-aux/snap/snapcraft.yaml [hunk 1]
```diff
@@ -203,9 +203,10 @@ parts:
       craftctl default
       # set version, this needs dpkg-parsechangelog (from dpkg-dev) and git
       VERSION="$(./mkversion.sh --output-only)"
-      if [ -f fips-build ] ; then
-          echo "-- appending FIPS tag to version $VERSION"
-          VERSION="$VERSION-fips"
+      if [ "${VERSION%/+fips/}" != "$VERSION" ] ; then
+          # we gave a '+fips' element in the version
```
Comment (zyga) [inline build-aux/snap/snapcraft.yaml:207]: Gave?

**Review by zyga — APPROVED** [review]
Bulk of the changes look good. I left a comment on a code comment that could use some clearer wording IMO.

LGTM when you want.

File: build-aux/snap/snapcraft.yaml [hunk 2]
```diff
@@ -203,10 +203,23 @@ parts:
...
-      if [ -f fips-build ] ; then
-          echo "-- appending FIPS tag to version $VERSION"
-          VERSION="$VERSION-fips"
+      if [ "${VERSION%/+fips/}" != "$VERSION" ] ; then
+          # we have a '+fips' element in the version, which may be coming from
+          # debian/changelog or git tag
+          echo "-- deteceted FIPS build"
+          touch fips-build
       fi
+      # TODO detect when doing a FIPS snap build on LP
```
Comment (zyga) [inline build-aux/snap/snapcraft.yaml:212]: Is there a ticket for that?

**Review by zyga — APPROVED** [review]

File: build-aux/snap/snapcraft.yaml (hunk 2 above)
Comment (bboozzoo) [reply build-aux/snap/snapcraft.yaml:212]: yup https://warthogs.atlassian.net/browse/SNAPDENG-23124

File: build-aux/snap/snapcraft.yaml [hunk 3]
```diff
@@ -203,10 +203,23 @@ parts:
       craftctl default
       # set version, this needs dpkg-parsechangelog (from dpkg-dev) and git
       VERSION="$(./mkversion.sh --output-only)"
-      if [ -f fips-build ] ; then
-          echo "-- appending FIPS tag to version $VERSION"
-          VERSION="$VERSION-fips"
+      if [ "${VERSION%/+fips/}" != "$VERSION" ] ; then
```
Comment (valentindavid) [inline build-aux/snap/snapcraft.yaml:206]: Did you mean `"${VERSION%+fips}"` or `"${VERSION/%+fips/}"`?

File: build-aux/snap/snapcraft.yaml [hunk 4]
```diff
@@ -203,10 +203,23 @@ parts:
...
+          # we have a '+fips' element in the version, which may be coming from
+          # debian/changelog or git tag
+          echo "-- deteceted FIPS build"
+          touch fips-build
       fi
+      # TODO detect when doing a FIPS snap build on LP
+
+      if [ -f fips-build ]; then
+          if [ "${VERSION%/+fips/}" = "$VERSION" ] ; then
```
Comment (valentindavid) [inline build-aux/snap/snapcraft.yaml:215]: Same here, did you mean "${VERSION%+fips}" or "${VERSION/%+fips/}"?

**Review by valentindavid — COMMENTED** [review]
A few comments

File: build-aux/snap/snapcraft.yaml (hunk 3 above)
Comment (bboozzoo) [reply build-aux/snap/snapcraft.yaml:206]: ah yes, too many changes going back and forth, `/+fips/` let me fix that

Comment (bboozzoo) [reply build-aux/snap/snapcraft.yaml]: it's just a simplification, debian/rules checks simply checks for +fips too, but we'll only ever append +fips as a suffix int eh changelog

**Review by valentindavid — APPROVED** [review]

**Comment by bboozzoo** [comment]
Failing tests:
- google:ubuntu-20.04-64:tests/main/snap-remove-terminate - known, unrelated

**Pull request merged**